src/Resources/config/services/services.xml

```diff
@@ -44,6 +44,18 @@
             <argument id="serializer.mapping.cache.symfony" type="service" />
         </service>
 
+        <service id="FINDOLOGIC\FinSearch\Controller\ProductDebugController" public="true">
```

Comment:
Maybe it is easier to add public getters for all properties in the main exporter and inject the `ExportController` as a service here. This way the debug controller isn't as strictly bound to the `ExportController` as it is right now, as basically all the logic needs to be called by the parent, which may lead to unexpected behavior.

Reply:
This would also lead to a lot of duplicated code, or a lot of public functions inside the ExportController.
I think it makes sense to add public getters.